src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceIdentifier.cs

```diff
@@ -32,11 +32,11 @@ public bool Equals(ServiceIdentifier other)
         {
             if (ServiceKey == null && other.ServiceKey == null)
             {
-                return ServiceType.Equals(other.ServiceType);
+                return ServiceType == other.ServiceType;
```

Comment:
`operator==` is overloaded for System.Type and it just forwards the call to `Equals` unless the type is RuntimeType: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Type.cs#L654-L666

Is it actually fixing anything?

Reply:
As long as you are trying to compare `RuntimeType` against `DecoratedType` at least. 

The comment 
``` csharp
// If `left` is a non-runtime type with a weird Equals implementation
// this is where operator `==` would differ from `Equals` call.
```

seems to indicate that someone else already experienced that...